packages/clerk-js/src/core/resources/SignIn.ts

```diff
@@ -451,6 +452,8 @@ export class SignIn extends BaseResource implements SignInResource {
       this.createdSessionId = data.created_session_id;
       this.userData = new UserData(data.user_data);
     }
+
+    eventBus.emit('signin:update', { resource: this });
```

Comment:
How would this look if instead of SignInResource, we moved it to BaseResource instead and changed the event name accordingly? 

Is there a specific reason to emit only for SignIn updates?